tests/integration/security/sds_tls_origination/util/util.go

```diff
@@ -0,0 +1,410 @@
+		secret := createSecret(credentialType, cn, systemNS.Name(), egressCred, isNotGeneric)
+		_, err := cluster.CoreV1().Secrets(systemNS.Name()).Create(context.TODO(), secret, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("Failed to create secret (error: %s)", err)
+		}
+	}
+	// Check if Kubernetes secret is ready
+	retry.UntilSuccessOrFail(t, func() error {
+		for _, cn := range credNames {
+			_, err := cluster.CoreV1().Secrets(systemNS.Name()).Get(context.TODO(), cn, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("secret %v not found: %v", cn, err)
+			}
+		}
+		return nil
+	}, retry.Timeout(time.Second*5))
+}
+
+// DeleteKubeSecret deletes a secret
+// nolint: interfacer
+func DeleteKubeSecret(t test.Failer, ctx framework.TestContext, credNames []string) {
+	// Get namespace for gateway pod.
+	istioCfg := istio.DefaultConfigOrFail(t, ctx)
+	systemNS := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+
+	if len(credNames) == 0 {
+		ctx.Log("no credential names are specified, skip creating ingress secret")
+		return
+	}
+	// Create Kubernetes secret for ingress gateway
+	cluster := ctx.Environment().(*kube.Environment).KubeClusters[0]
+	for _, cn := range credNames {
+		var immediate int64
+		err := cluster.CoreV1().Secrets(systemNS.Name()).Delete(context.TODO(), cn,
+			metav1.DeleteOptions{GracePeriodSeconds: &immediate})
+		if err != nil {
+			t.Fatalf("Failed to create secret (error: %s)", err)
+		}
+	}
+}
+
+// createSecret creates a kubernetes secret which stores CA cert for SIMPLE TLS.
+// For mTLS ingress gateway, createSecret adds client cert and key into the secret object.
+func createSecret(credentialType string, cn, ns string, ic TLSCredential, isNotGeneric bool) *v1.Secret {
+	if credentialType == "MUTUAL" {
+		if isNotGeneric {
+			return &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cn,
+					Namespace: ns,
+				},
+				StringData: map[string]string{
+					tlsScrtCert:   ic.ClientCert,
+					tlsScrtKey:    ic.PrivateKey,
+					tlsScrtCaCert: ic.CaCert,
+				},
+			}
+		}
+		return &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cn,
+				Namespace: ns,
+			},
+			StringData: map[string]string{
+				genericScrtCert:   ic.ClientCert,
+				genericScrtKey:    ic.PrivateKey,
+				genericScrtCaCert: ic.CaCert,
+			},
+		}
+	}
+	if isNotGeneric {
+		return &v1.Secret{}
+	}
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cn,
+			Namespace: ns,
+		},
+		StringData: map[string]string{
+			genericScrtCaCert: ic.CaCert,
+		},
+	}
+}
+
+// SetupEcho creates two namespaces client and server. It also brings up two echo instances server and
+// client in respective namespaces. HTTP and HTTPS port on the server echo are set up. Egress Gateway is set up in the
+// service namespace to handle egress for "external" calls.
+func SetupEcho(t *testing.T, ctx resource.Context) (echo.Instance, echo.Instance, namespace.Instance, namespace.Instance) {
```

Comment:
This setup is similar to one in security/filebased_tls_origination/, it's better to share the setup util functions. Could be done in a separate PR.

Reply:
There's some significant differences between the two, most probably I need to change the SetupEcho for that function first and then I can maybe combine